lxd/apparmor/qemuimg.go

```diff
@@ -0,0 +1,150 @@
+package apparmor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+var qemuImgProfileTpl = template.Must(template.New("qemuImgProfile").Parse(`#include <tunables/global>
+profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  /usr/bin/prlimit    mrix,
+  /usr/bin/qemu-img   mrix,
+  {{ .pathToImg }} rk,
+ 
+}
+`))
+
+type nullWriteCloser struct {
+	*bytes.Buffer
+}
+
+func (nwc *nullWriteCloser) Close() error {
+	return nil
+}
+
+func RunQemuImgWithApparmor(sysOS *sys.OS, imgPath string) (string, error) {
+	imgJSON, err := runQemuImgWithApparmor("prlimit", []string{"--cpu=2", "--as=1000000000", "qemu-img", "info", "-f", "qcow2", "--output=json", imgPath}, nil, sysOS, imgPath)
+	return imgJSON, err
+}
+
+func runQemuImgWithApparmor(cmd string, args []string, stdin io.ReadCloser, sysOS *sys.OS, imgPath string) (string, error) {
```

Comment:
The `WithApparmor` is implicit given its in the `apparmor` package, so can just be `QemuImg()` I think.